js/attributePoints.js

```diff
@@ -3,73 +3,26 @@ let listaCheckbox = document.querySelectorAll("input[type='checkbox']");
-
-        let verificouCheckboxAtual = false;
-        let temMaisPontosAtivos = false;
-        let habilitarPontosAdicionais = false;
-        let novaPontuacao = 0;
+        let novaPontuacao;
         
         //Verifica a nova pontuação que será atribuída
-        for (let i = listaCheckboxAtual.length - 1; i >= 0; i--)
-        {
-            if (verificouCheckboxAtual)
-            {
-                if (!listaCheckboxAtual[i].checked)
-                    novaPontuacao++;                   
-            }
-            else if (listaCheckboxAtual[i] == this)
-            {
-                if (listaCheckboxAtual[i].checked)
-                    novaPontuacao++;
-                else if (!temMaisPontosAtivos)
-                    novaPontuacao--;                    
-                verificouCheckboxAtual = true;               
-            }            
-            else
-            {
-                if (listaCheckboxAtual[i].checked)
-                    novaPontuacao--;                   
-                temMaisPontosAtivos = listaCheckboxAtual[i].checked;                                     
-            }
-        }
+        if (!this.checked && this.value == pontuacaoAtributos.getPontos())
+            novaPontuacao = - 1;
+        else
+            novaPontuacao = this.value - pontuacaoAtributos.getPontos();
 
         //Se a nova pontuação for válida, preenche os pontos na página
         if (pontuacaoAtributos.adicionarPontos(novaPontuacao)){
-            verificouCheckboxAtual = false;
-            temMaisPontosAtivos = false;            
-            for (let i = listaCheckboxAtual.length - 1; i >= 0; i--)
-            {
-                if (verificouCheckboxAtual)
-                {
-                    listaCheckboxAtual[i].checked = true;            
-                }
-                else if (listaCheckboxAtual[i] == this)
-                {
-                    if (temMaisPontosAtivos)
-                    {
-                        listaCheckboxAtual[i].checked = true;
-                    }
-                    else if (listaCheckboxAtual[i].checked)
-                    {
-                        habilitarPontosAdicionais = (i == listaCheckboxAtual.length - 1);                
-                    }
-                    verificouCheckboxAtual = true;               
-                }            
-                else
-                {                   
-                    temMaisPontosAtivos = listaCheckboxAtual[i].checked;
-                    listaCheckboxAtual[i].checked = false;                                      
-                }            
-            }
+            let listaCheckboxAtual = this.parentNode.querySelectorAll("input[type='checkbox']");
+            
+            listaCheckboxAtual.forEach(checkbox => {
+                checkbox.checked = (checkbox.value <= pontuacaoAtributos.getPontos());
+            });

```

Comment:
Daria para fazer em 1 linha, embora comprometa a legibilidade.
```suggestion
            listaCheckboxAtual.forEach(checkbox => checkbox.checked = (checkbox.value <= pontuacaoAtributos.getPontos()));
```